purchase_manual_service_quantity/models/purchase_order_line.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PurchaseOrderLine(models.Model):
+
+    _inherit = 'purchase.order.line'
+
+    @api.depends('order_id.state', 'move_ids.state')
+    def _compute_qty_received(self):
+        for line in self:
+            if line.product_id.type in ['service'] and\
+                    line.product_id.purchase_manual_received_qty:
+                line.qty_received = line.qty_received_manually
+            else:
+                super(PurchaseOrderLine, self)._compute_qty_received()
+
+    def _set_qty_received(self):
+        for line in self:
+            if line.product_id.type in ['service'] and \
+                    line.product_id.purchase_manual_received_qty:
```

Comment:
I think there should be a method or a computed field which tell if we can manually set received quantity on the line. Actually, there is the same condition in compute and set method.